bsp4s/src/main/scala/ch/epfl/scala/bsp/Bsp.scala

```diff
@@ -576,6 +576,23 @@ object TestParamsDataKind {
     targets: List[BuildTargetIdentifier]
 )
 
+
+@JsonCodec final case class ExecutionEnvironmentParams(
+    targets: List[BuildTargetIdentifier],
+    originId: Option[String]
+)
+
+@JsonCodec final case class ClasspathItem(
+    target: BuildTargetIdentifier,
```

Comment:
Is the purpose of this abstraction to represent a classpath per build target? If so, the name should make that clearer, `ClasspathItem` suggests a `List[String]`.
  
In addition to that, the classpath is leaking some key information about build targets which could be considered internal. What happens if this classpath goes out-of-date from one compilation to another one? If we don't handle this use case, are we not assuming build tools can always generate a unique classpath that doesn't change? This assumption might be wrong if, for example, a build tool generates immutable jars after every compilation.

Reply:
> Is the purpose of this abstraction to represent a classpath per build target? If so, the name should make that clearer, ClasspathItem suggests a List[String].
Yes, that was right before I added jvmOptions to the class and renamed it. Now I think a dedicated case class for this is needed (`JvmEnvironmentEntry`)

> In addition to that, the classpath is leaking some key information about build targets which could be considered internal. What happens if this classpath goes out-of-date from one compilation to another one? If we don't handle this use case, are we not assuming build tools can always generate a unique classpath that doesn't change after compilation runs? This assumption might be wrong if, for example, a build tool generates immutable jars after every compilation.
My assumption is that, the IDE is doing this call right before the execution, but after compilation is finished, so it's up to the server to keep it up-to-date. Please correct me if I'm wrong.

Reply:
> What happens if this classpath goes out-of-date from one compilation to another one? 

I think you can treat it the same way `buildTargets/scalacOptions` already leaks internal information. IntelliJ and Metals are already using those values to assemble a classpaths. I believe the reason it's working currently for IntelliJ is that it uses a custom stable classes directory, which Bloop does not remove after the BSP session closes.

I don't know what the best solution is for manifest jars however. Maybe it's OK to create temporary files and leave them there forever 🤔 Maybe we should include in the `jvmTestEnvironment` metadata about whether the client should delete the classpath entries?

Reply:
> My assumption is that, the IDE is doing this call right before the execution, but after compilation is finished, so it's up to the server to keep it up-to-date.

That sounds good, so long as we make this clear in the spec that the client should not cache this at any cost, then my concern is resolved. BTW, what do you think about updating the spec with these changes?

> I think you can treat it the same way buildTargets/scalacOptions already leaks internal information

Yup, my main concern is about caching and how the client should handle that, especially in the presence of manifest jars that are immutable (e.g. bazel). However, if this call is done right after every compilation, this issue should be resolved. The problem comes when the jars in the classpath needs to survive several compilation runs since they assume mutability.

Reply:
> BTW, what do you think about updating the spec with these changes?

Of course, just doing it